example/src/Examples/Graphs/Interpolation.tsx

```diff
@@ -1,66 +1,121 @@
+import type {
+  AnimatedProps,
+  PathDef,
+  PathProps,
+  SkPath,
+} from "@shopify/react-native-skia";
 import {
+  processPath,
+  Easing,
   Canvas,
   Fill,
-  LinearGradient,
   Path,
-  runSpring,
-  Spring,
+  runTiming,
   useDerivedValue,
   useValue,
-  vec,
 } from "@shopify/react-native-skia";
-import React, { useCallback, useEffect, useMemo, useState } from "react";
+import React, { useCallback, useEffect, useRef, useState } from "react";
 import { StyleSheet, Text, View } from "react-native";
 
 import { createGraphPath } from "./createGraphPath";
 import type { GraphProps } from "./types";
 
 export const Interpolation: React.FC<GraphProps> = ({ height, width }) => {
-  const path = useMemo(
-    () => createGraphPath(width, height, 60),
-    [height, width]
-  );
-
-  const path2 = useMemo(
-    () => createGraphPath(width, height, 60),
-    [height, width]
+  const [currentPath, setCurrentPath] = useState<SkPath>(() =>
+    createGraphPath(width, height, 60)
   );
 
-  const progress = useValue(0);
-  const [toggled, setToggled] = useState(false);
-  const onPress = useCallback(() => setToggled((p) => !p), []);
-  useEffect(() => {
-    runSpring(progress, toggled ? 1 : 0, Spring.Config.Gentle);
-  }, [progress, toggled]);
-
-  const interpolatedPath = useDerivedValue(
-    () => path.interpolate(path2, progress.current),
-    [progress]
-  );
+  const onPress = useCallback(() => {
+    setCurrentPath(createGraphPath(width, height, 60));
+  }, [height, width]);
 
   return (
     <View style={{ height, marginBottom: 10 }} onTouchEnd={onPress}>
       <Canvas style={styles.graph}>
         <Fill color="black" />
-        <Path
-          path={interpolatedPath}
+        <InterpolatedPath
+          path={currentPath}
           strokeWidth={4}
           style="stroke"
           strokeJoin="round"
           strokeCap="round"
-        >
-          <LinearGradient
-            start={vec(0, height * 0.5)}
-            end={vec(width * 0.5, height * 0.5)}
-            colors={["black", "#cccc66"]}
-          />
-        </Path>
+          color="#cccc66"
+        />
       </Canvas>
       <Text>Touch graph to interpolate</Text>
     </View>
   );
 };
 
+const InterpolatedPath = ({
```

Comment:
That's a super nice approach.
InterpolatedPath may be confusing, this is more of a PathTransition or a PathWithTransition or a TransitionedPath or something? 🤔